app/services/websocket.js

```diff
@@ -0,0 +1,33 @@
+import PhoenixSocket from 'phoenix/services/phoenix-socket';
+
+export default PhoenixSocket.extend({
+  websocketServerUrl: `ws://${document.location.hostname}:4000`,
+  channel: null,
+
+  connect() {
```

Comment:
I'm listening to 'open' and 'close' events forever, how would I accomplish that with a one-off Promise ?